web/vtadmin/src/components/routes/shard/Advanced.tsx

```diff
@@ -0,0 +1,413 @@
+import React, { useState } from 'react';
+import { useParams, Link, useHistory } from 'react-router-dom';
+import {
+    useDeleteShard,
+    useEmergencyFailoverShard,
+    useKeyspace,
+    usePlannedFailoverShard,
+    useReloadSchemaShard,
+    useTabletExternallyPromoted,
+    useTablets,
+} from '../../../hooks/api';
+import { useDocumentTitle } from '../../../hooks/useDocumentTitle';
+import ActionPanel from '../../ActionPanel';
+import { success, warn } from '../../Snackbar';
+import { UseMutationResult } from 'react-query';
+import Toggle from '../../toggle/Toggle';
+import { Label } from '../../inputs/Label';
+import { TextInput } from '../../TextInput';
+import { NumberInput } from '../../NumberInput';
+import { Select } from '../../inputs/Select';
+import { formatAlias, formatDisplayType } from '../../../util/tablets';
+import { logutil, vtadmin } from '../../../proto/vtadmin';
+import Dialog from '../../dialog/Dialog';
```

Comment:
We'll probably switch to non-relative paths at some point down the line.

Reply:
Yeah, I think that would be really helpful in cleaning things up.